src/floats.jl

```diff
@@ -45,31 +75,30 @@ bits(::Type{T}) where {T <: Union{Float16, Float32, Float64}} = 8sizeof(T)
         end
     end
     v == 0 && return zero(T)
-    # if v < 2ms
-    #     if 0 <= exp < 2cl
-    #         return T(Base.twiceprecision(Base.TwicePrecision{T}(v) * pow10(T, exp), significantbits(T)))
-    #     elseif -2cl < exp < 0
-    #         return T(Base.twiceprecision(Base.TwicePrecision{T}(v) / pow10(T, -exp), significantbits(T)))
-    #     end
-    # end
-    mant = big(v)
+    mant = BigInt!(NUM, v)
     if 0 <= exp < 327
-        num = mant * bipows5[exp+1]
+        num = MPZ.mul!(mant, bipows5[exp+1])
         bex = bitlength(num) - significantbits(T)
         bex <= 0 && return ldexp(T(num), exp)
-        quo = roundQuotient(num, big(1) << bex)
+        MPZ.mul_2exp!(MPZ.set_si!(ONE, 1), bex)
+        quo = roundQuotient(num, ONE)
```

Comment:
Just so I understand: this is essentially implementing `Float64(x::BigInt)`?

Man, we need to improve the performance of the one in Base.

Reply:
https://github.com/JuliaLang/julia/issues/31293